src/content/popup/Metadata/EraInfoComponent.tsx

```diff
@@ -1,29 +1,97 @@
-import type { EraInfo, EraMeta } from '../../years';
+import { useMemo, useState } from 'preact/hooks';
+
+import { useLocale } from '../../../common/i18n';
+import { classes } from '../../../utils/classes';
+
+import {
+  type EraInfoDate,
+  type EraInfoTimeSpan,
+  type EraMeta,
+  getEraInfoTimeSpan,
+} from '../../dates';
 
 type Props = {
   meta: EraMeta;
-  eraInfo: EraInfo;
 };
 
 export function EraInfoComponent(props: Props) {
-  const seireki =
-    props.meta.year === 0
-      ? props.eraInfo.start
-      : props.meta.year - 1 + props.eraInfo.start;
+  const { t } = useLocale();
+
+  const [timeSpan, setTimeSpan] = useState<EraInfoTimeSpan | undefined>(
+    undefined
+  );
+
+  useMemo(() => {
+    void getEraInfoTimeSpan(props.meta).then(setTimeSpan);
+  }, [props.meta]);
```

Comment:
I think this is a _little_ bit weird in React terms since I think we'll do an initial render showing the "No result" string before updating. However, I think those two renders will happen before the browser actually presents a frame so the user won't notice it.

If we really wanted to do things properly, we'd probably make it the component tri-state: updating, updated, and error but it's probably not worth it given that the user won't see any difference.

Reply:
Yes, I see your point. However, because of the aspects you mentioned, I think this is one of the cases where the simple, naive approach is good enough.
Instead of an updating state I'd rather just cache the last calculated time span and display that. That way we would also get rid of the flickering.

Reply:
> Instead of an updating state I'd rather just cache the last calculated time span and display that. That way we would also get rid of the flickering.

Seems to work. Once you verify this, I won’t have any further concerns about merging the PR :)